### ISSUE TITLE
Replace album with a generic tag concept

### ISSUE THREAD

**Issue by ttyridal** [photoview/photoview#833]
The most promising self hosted gallery out there! Thank you for your efforts

I'd like to propose replacing/extending the album concept to include more generic tags.

On import, the folder to album mapping would be kept. Additionally it would be possible to read exif data or a .photoview-tags special file for more tags

The UI should have a way to add/remove tags on a photo (ie create/manage albums), preferably with multi-select

Sidebar should list all tags/albums the image is part of (much like faces work)

This would also make it easy to fit auto tagging (object detection/scene detection) in the future. Could perhaps be an external service?

**Comment by Leonetienne** [comment]
I would also love the Idea of tags. Pr0gramm uses tags and you can easily find an 8yr old image (thousands of posts per day) if you know what it looked like.

However, I would also absolutely love to see nested albums. I have seen no gallery app, not even just a mobile gallery app for local pictures, offering this.

Nested Albums would work kind of like a directory tree, having `Vacation/car.jpg`,  `Vacation/Berlin/*.jpg` `Vacation/France/*.jpg`, `Family/Maria/*.jpg`, etc. That would allow having the same picture of Maria both in `Vacation/France`, and `Family/Maria` without having to store it twice, or making a symlink. Sure, this could also be achieved by using tags, but this is a more "browse-friendly" approach if you're not looking for a specific image. I am considering making my own gallery app just for this feature, because nobody seems wanting to implement it.

**Comment by ttyridal** [comment]
Indeed, i also maintain a folder structure. Even implemented a simple gallery for it :)  my conclusion was that I'd rather have it as a flat view and use naming convention for grouping:
Vacation/X
Vacation/Y
Etc

I guess it depends on the size of your gallery 🤷🏼‍♂️

The tags/album overview could have multiple display options: list, small icons, large icons etc

You did capture a key point that I didn't wrote though: the ability to (logically) have a photo in multiple groups/tags/albums, with only one physical copy on disk.

**Comment by kkovaletp** [comment]
> That would allow for the tidyness of a directory tree, while having the same picture of Maria both in `Vacation/France/`, and `Family/Maria/` without having to store it twice

@Leonetienne, as Photoview relies on the file system and replicates 1:1 what is there to Albums, we should not introduce any difference in this logic. However, your request still can be done on the file system level by creating hard or soft links for the same file in different folders.

Make sure to not create dead loops of links.

**Comment by kkovaletp** [comment]
I like the idea of a virtual grouping of media in addition to the physical hierarchy (represented by Albums).
To not complicate things and keep the UX for users clear and transparent, we shouldn't modify the logic and functionality of Albums by adding some tags or similar things inside. Instead, it would be much better to add a new section (like Categories), which allows users to create their own structure of tags and show media in that structure according to the tags assigned.
Assigning tags can be done in the media info panel in a similar way to sharing functionality.

**Comment by kkovaletp** [comment]
FYI: Here is a discussion #1283 about a similar use case
